### ISSUE TITLE
PlanBuild Not working with Ashlands update?

### ISSUE THREAD

**Issue by evexavier** [sirskunkalot/PlanBuild#90]
[Info   :PlanBuild.Plans.PlanManager] Initializing PlanManager
[Error  : Unity Log] InvalidOperationException: Parameter #0 of hook for Void Destroy(HitData) doesn't match, must be WearNTear or related
Stack trace:
MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target, MonoMod.RuntimeDetour.HookConfig& config) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target, MonoMod.RuntimeDetour.HookConfig config) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase method, System.Delegate to) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpoint._NewHook (System.Reflection.MethodBase from, System.Delegate to) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpoint._Add[TDelegate] (System.Func`3[T1,T2,TResult] gen, TDelegate hookDelegate) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpoint.Add (System.Delegate hookDelegate) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpointManager.Add (System.Reflection.MethodBase method, System.Delegate hookDelegate) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpointManager.Add[T] (System.Reflection.MethodBase method, System.Delegate hookDelegate) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
On.WearNTear.add_Destroy (On.WearNTear+hook_Destroy ) (at <357d26bb0e44465093c0e8536dd2e067>:0)
PlanBuild.Plans.PlanManager.Init () (at <2ad234d4a05e4c82816b58eeab1f12d2>:0)
PlanBuild.PlanBuildPlugin.Awake () (at <2ad234d4a05e4c82816b58eeab1f12d2>:0)
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.GameObject:.cctor()
PlatformTools.Common.SingletonMonoBehaviour`1:get_Instance()
PlatformTools.PlatformManagerInitializer:OnRuntimeMethodLoad()

**Comment by MWGMorningwood** [comment]
Breaking for me as well

```
[Info   :   BepInEx] Loading [PlanBuild 0.15.0]
[Info   :Jotunn.Main] Initializing InputManager
[Error  : Unity Log] MissingFieldException: Field not found: int .Piece.ghostLayer Due to: Could not find field in class
Stack trace:
(wrapper dynamic-method) Piece.DMD<Piece::Awake>(Piece)
UnityEngine.Object:Instantiate(GameObject)
PlanBuild.Blueprints.BlueprintAssets:Load(AssetBundle)
PlanBuild.PlanBuildPlugin:Awake()
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.GameObject:.cctor()
PlatformTools.Common.SingletonMonoBehaviour`1:get_Instance()
PlatformTools.PlatformManagerInitializer:OnRuntimeMethodLoad()

[Info   :PlanBuild.Blueprints.BlueprintManager] Initializing BlueprintManager
[Info   :Jotunn.Main] Initializing CommandManager
[Info   :PlanBuild.Blueprints.BlueprintSync] Loading known blueprints
[Info   :Jotunn.Main] Initializing UndoManager
[Warning:PlanBuild.Blueprints.BlueprintManager] Error caught while initializing: System.InvalidProgramException: Invalid IL code in (wrapper dynamic-method) Player:DMD<Player::UpdatePlacementGhost> (Player,bool): IL_048d: call      0x000000f5


  at (wrapper managed-to-native) System.RuntimeMethodHandle.GetFunctionPointer(intptr)
  at System.RuntimeMethodHandle.GetFunctionPointer () [0x00000] in <17d9ce77f27a4bd2afb5ba32c9bea976>:0
  at MonoMod.RuntimeDetour.Platforms.DetourRuntimeILPlatform.GetFunctionPointer (System.Reflection.MethodBase method, System.RuntimeMethodHandle handle) [0x00000] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.Platforms.DetourRuntimeILPlatform.GetNativeStart (System.Reflection.MethodBase method) [0x0004d] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.DetourHelper.GetNativeStart (System.Reflection.MethodBase method) [0x00005] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.Detour._RefreshChain (System.Reflection.MethodBase method) [0x000fb] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.Detour.Apply () [0x00053] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.Hook.Apply () [0x00054] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target, MonoMod.RuntimeDetour.HookConfig& config) [0x0049d] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target, MonoMod.RuntimeDetour.HookConfig config) [0x00000] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target) [0x0001d] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase method, System.Delegate to) [0x0000e] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.HookGen.HookEndpoint._NewHook (System.Reflection.MethodBase from, System.Delegate to) [0x00000] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.HookGen.HookEndpoint._Add[TDelegate] (System.Func`3[T1,T2,TResult] gen, TDelegate hookDelegate) [0x0003c] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.HookGen.HookEndpoint.Add (System.Delegate hookDelegate) [0x00000] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.HookGen.HookEndpointManager.Add (System.Reflection.MethodBase method, System.Delegate hookDelegate) [0x00028] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at MonoMod.RuntimeDetour.HookGen.HookEndpointManager.Add[T] (System.Reflection.MethodBase method, System.Delegate hookDelegate) [0x00000] in <4e2760c7517c4ea79c633d67e84b319f>:0
  at On.Player.add_UpdatePlacementGhost (On.Player+hook_UpdatePlacementGhost ) [0x0000a] in <6133f85798694298866a831d6b53e741>:0
  at PlanBuild.Blueprints.BlueprintManager.Init () [0x000aa] in <2ad234d4a05e4c82816b58eeab1f12d2>:0
[Info   :Jotunn.Main] Initializing KitbashManager
[Info   :PlanBuild.Plans.PlanManager] Initializing PlanManager
[Error  : Unity Log] InvalidOperationException: Parameter #0 of hook for Void Destroy(HitData) doesn't match, must be WearNTear or related
Stack trace:
MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target, MonoMod.RuntimeDetour.HookConfig& config) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target, MonoMod.RuntimeDetour.HookConfig config) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase from, System.Reflection.MethodInfo to, System.Object target) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.Hook..ctor (System.Reflection.MethodBase method, System.Delegate to) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpoint._NewHook (System.Reflection.MethodBase from, System.Delegate to) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpoint._Add[TDelegate] (System.Func`3[T1,T2,TResult] gen, TDelegate hookDelegate) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpoint.Add (System.Delegate hookDelegate) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpointManager.Add (System.Reflection.MethodBase method, System.Delegate hookDelegate) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
MonoMod.RuntimeDetour.HookGen.HookEndpointManager.Add[T] (System.Reflection.MethodBase method, System.Delegate hookDelegate) (at <4e2760c7517c4ea79c633d67e84b319f>:0)
On.WearNTear.add_Destroy (On.WearNTear+hook_Destroy ) (at <6133f85798694298866a831d6b53e741>:0)
PlanBuild.Plans.PlanManager.Init () (at <2ad234d4a05e4c82816b58eeab1f12d2>:0)
PlanBuild.PlanBuildPlugin.Awake () (at <2ad234d4a05e4c82816b58eeab1f12d2>:0)
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.GameObject:.cctor()
PlatformTools.Common.SingletonMonoBehaviour`1:get_Instance()
PlatformTools.PlatformManagerInitializer:OnRuntimeMethodLoad()
```

**Comment by sirskunkalot** [comment]
Fixed in v0.16.0

**Comment by evexavier** [comment]
Thanx so much. Love this mod!
